File: CS/SchedulerGSync/SchedulerGSync/Helpers/ColorHelper.cs
using System;
using System.Drawing;
using System.Globalization;
using System.Linq;

namespace SchedulerGSync {
    public static class ColorHelper {
        public static Color FromString(string stringValue) {
            if (stringValue.StartsWith("#")) {
                if (stringValue.Length == 7) {
                    string colorHexValue = stringValue.Substring(1, 6);
                    int r = int.Parse(stringValue.Substring(1, 2), NumberStyles.HexNumber);
                    int g = int.Parse(stringValue.Substring(3, 2), NumberStyles.HexNumber);
                    int b = int.Parse(stringValue.Substring(5, 2), NumberStyles.HexNumber);
                    return Color.FromArgb(r, g, b);
                }
                return Color.Pink;
            } else
                return Color.FromName(stringValue);
        }
    }
}

File: VB/SchedulerGSync/SchedulerGSync/Helpers/ColorHelper.vb
Imports System
Imports System.Drawing
Imports System.Globalization
Imports System.Linq

Namespace SchedulerGSync
	Public Module ColorHelper
		Public Function FromString(ByVal stringValue As String) As Color
			If stringValue.StartsWith("#") Then
				If stringValue.Length = 7 Then
					Dim colorHexValue As String = stringValue.Substring(1, 6)
					Dim r As Integer = Integer.Parse(stringValue.Substring(1, 2), NumberStyles.HexNumber)
					Dim g As Integer = Integer.Parse(stringValue.Substring(3, 2), NumberStyles.HexNumber)
					Dim b As Integer = Integer.Parse(stringValue.Substring(5, 2), NumberStyles.HexNumber)
					Return Color.FromArgb(r, g, b)
				End If
				Return Color.Pink
			Else
				Return Color.FromName(stringValue)
			End If
		End Function
	End Module
End Namespace
